test/functional/test_framework/test_node.py

```diff
@@ -313,25 +316,35 @@ def wait_for_rpc_connection(self, *, wait_for_import=True):
                 return
             except JSONRPCException as e:  # Initialization phase
                 # -28 RPC in warmup
+                if e.error['code'] == -28:
+                    ignored_errors["JSONRPCException -28"] += 1
                 # -342 Service unavailable, RPC server started but is shutting down due to error
-                if e.error['code'] != -28 and e.error['code'] != -342:
+                # - Ignore the exception to properly raise the FailedToStartError.
+                elif e.error['code'] == -342:
+                    ignored_errors["JSONRPCException -342"] += 1
```

Comment:
non-blocking nit: might be a bit more Pythonic to use Counter.update() rather than `+= 1`.
For example:
```python
ignored_errors.update([f"JSONRPCException {e.error['code']}"])
```

Reply:
Tried using `update()`, but it resulted in things like:
```
AssertionError: [node 0] Unable to connect to bitcoind after 10s (ignored errors: {'m': 40, 'i': 120, 's': 120, 'n': 80, 'g': 40, '_': 40, 'c': 40, 'r': 40, 'e': 80, 'd': 40, 't': 40, 'a': 40, 'l': 40}, latest error: ValueError('No RPC credentials'))
```
i.e. it was iterating over the string and incrementing an entry for each unique character, which seems to match the [docs](https://docs.python.org/3/library/collections.html#collections.Counter.update) even though not super clear.